File: biner.py
# Program Angka Biner (Konversi Bilangan Bulat ke Biner)

# Menerima input dari user
angka = int(input("Masukkan bilangan bulat (>= 0): "))

# Validasi input
if angka < 0:
    print("Input tidak valid! Masukkan bilangan >= 0.")
else:
    # Konversi ke biner (tanpa prefix '0b')
    biner = bin(angka)[2:]
    print(f"Bilangan {angka} dalam biner adalah: {biner}")

File: week2/biner.dart
import 'dart:io';

// Fungsi untuk mengubah bilangan bulat ke string biner
String toBinary(int n) {
  if (n < 0) {
    throw ArgumentError("N harus >= 0");
  }
  return n.toRadixString(2); // konversi ke biner
}

void main() {
  stdout.write("Masukkan bilangan bulat: ");
  int? N = int.tryParse(stdin.readLineSync()!);

  if (N == null || N < 0) {
    print("Input tidak valid. Harus berupa bilangan bulat >= 0.");
    return;
  }

  print("Bentuk biner dari $N = ${toBinary(N)}");
}
